doc/ApiDocGeneration/Generate-Api-Docs.ps1

```diff
@@ -0,0 +1,305 @@
+    "frontdoor"="Front Door"; "graphrbac"="Active Directory";
+    "guestconfiguration"="Guest Configuration"; "hdinsight"="HDInsight";
+    "healthcareapis"="Healthcare APIs"; "hybridcompute"="Hybrid Cloud";
+    "hybriddatamanager"="Hybrid Datamanager"; "identity"="Identity";
+    "insights"="Insights"; "iotcentral"="IoT Central";
+    "iothub"="IoT"; "keyvault"="Keyvault";
+    "kusto"="Kusto"; "labservices"="LabServices";
+    "locationbasedservices"="Location Based Services"; "logic"="Logic Apps";
+    "machinelearning"="Machine Learning"; "machinelearningcompute"="Machine Learning Compute";
+    "maintenance"="Maintenance"; "managednetwork"="Managed Network";
+    "managedserviceidentity"="Managed Service Identity"; "managedservices"="Managed Services";
+    "managementgroups"="Management Groups"; "managementpartner"="Management Partner";
+    "maps"="Maps"; "marketplaceordering"="Market Place Ordering";
+    "mediaservices"="Media Services";
+    "mixedreality"="Mixed Reality"; "monitor"="Monitor";
+    "netapp"="NetApp"; "network"="Network";
+    "notificationhubs"="Notification Hubs"; "operationalinsights"="Operational Insights";
+    "peering"="Peering"; "policyinsights"="Policy Insights";
+    "postgresql"="PostgreSQL"; "powerbidedicated"="PowerBI Dedicated";
+    "privatedns"="Private DNS"; "recoveryservices"="Recovery Services";
+    "recoveryservices-backup"="Recovery Services Backup"; "recoveryservices-siterecovery"="Recovery Services Site Recovery";
+    "redis"="Redis"; "relay"="Relay";
+    "reservations"="Reservations"; "resourcegraph"="Resource Graph";
+    "resources"="Resources"; "scheduler"="Scheduler";
+    "search"="Search"; "securitycenter"="Security Center";
+    "servermanagement"="Server Management"; "servicebus"="Service Bus";
+    "servicefabric"="Service Fabric"; "signalr"="signalR";
+    "sqlmanagement"="SQL Management"; "sqlvirtualmachine"="SQL Virtual Machine";
+    "storage"="Storage"; "storagecache"="Storage Cache";
+    "storagesync"="Storage Sync"; "storsimple"="Stor Simple";
+    "storsimple8000series"="Stor Simple 8000 series"; "streamanalytics"="Stream Analytics";
+    "subscription"="Subscription";
+    "trafficmanager"="Traffic Manager"; "websites"="Websites";
+}
+
+Function Get-DocFx
+{
+    [CmdletBinding()]
+    Param ($DownloadDirectory)
+
+    try 
+    {
+        Write-Verbose "Downloading DocFx..."
+        Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+        -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "$($DownloadDirectory)/docfx/"
+    }
+    catch 
+    {
+        Write-Verbose "Failed to Download DocFx"
+        Write-Error "$_"
+        Exit 1
+    }
+}
+
+Function Get-DocTools
+{
+    [CmdletBinding()]
+    Param (
+        $DownloadDirectory,
+        $RepoRoot
+    )
+
+    try 
+    {
+        Write-Verbose "Download and Extract mdoc to Build.BinariesDirectory/mdoc"
```

Comment:
The more I think about it the more I think the DocFx and DocTools should be pulled out an be a separate boot-strapping step in the pipelines. That ensure we can track when the fail and generally if you are repro'ing locally you only want to do these parts once and then iterate on the doc generation.

Reply:
Ok, that makes sense. I will update this.